### ISSUE TITLE
Optimize imports

### ISSUE THREAD

**Issue by danielgafni** [yukinarit/pyserde#604]
I've noticed `pyserde` contributes a fair share to my CLI app import time flamegraph (I'm using `pyserde` for app configuration so it has to be imported at top level). 

Here is the breakdown for `pyserde`: 

![image](https://github.com/user-attachments/assets/2ac335e0-aba6-40b4-abc8-9010cf50b17b)

It's really nothing to worry about, but I guess dependencies like `numpy` and `sqlalchemy` could be imported in a lazy fashion in order to further improve import times.  

This flamegraph can be made with [tuna](https://github.com/nschloe/tuna).

**Comment by yukinarit** [comment]
Hi @danielgafni! 

Thanks for making a framegraph! I've never done this optimization. 

Is it like moving all the imports to the code using it?

```python
        elif is_numpy_array(c):
            from .numpy import deserialize_numpy_array_direct
            res = deserialize_numpy_array_direct(c, o)
```

**Comment by danielgafni** [comment]
Yes exactly. If you need them for type annotations, you can do the following:

```python
from typing TYPE_CHECKING

if TYPE_CHECKING:  # will NOT be executed at runtime
    import numpy as np

def my_func(arg: "np.array"):
    ...
```

**Comment by yukinarit** [comment]
Looks good!

Would you be able to contribute to pyserde? :slightly_smiling_face:

**Comment by danielgafni** [comment]
Maybe! I'm short on time right now but it should improve in a few weeks

**Comment by yukinarit** [comment]
awesome!
